src/code/z_bgcheck.c

```diff
@@ -322,40 +320,34 @@ s32 CollisionPoly_CheckYIntersectApprox1(CollisionPoly* poly, Vec3s* vtxList, f3
  * Checks if point (`x`,`z`) is within `checkDist` of `poly`, computing `yIntersect` if true
  * Determinant max 0.0f (checks if on or within poly)
  */
-#ifdef NON_MATCHING
 s32 CollisionPoly_CheckYIntersect(CollisionPoly* poly, Vec3s* vtxList, f32 x, f32 z, f32* yIntersect, f32 checkDist) {
-    static Vec3f polyVerts[3]; // D_801EDA18
     Vec3s* sVerts;
```

Comment:
The concern I have with this is that the way the function was written previously had correct codegen, but didn't match because of .bss ordering issues; it seems likely that an in-function static declaration was used, it's just we don't know how to force the proper order for w/e reason.

Reply:
What was the motivation for having an in-function static again? This function matches just fine without in-function static, and making it in static as you mentioned puts the bss in the wrong order. This bss variable has to be above `D_801EDA40` and below `D_801ED9F0`

Reply:
mainly just the code quality improvements from keeping the variables within their scope of usage. It gets harder to name D_801ED9F0 and D_801EDA18 when they're defined at the file level. I guess the change is fine for now, but I can't help but feel that there's must be some trick to inline these and get the correct bss ordering.

Reply:
I'll actually go ahead and leave a comment with `D_801EDA18` stating that it can be in-function static bss much later down the road